Sources/SwiftKafka/RDKafka/RDKafkaConfig.swift

```diff
@@ -0,0 +1,103 @@
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crdkafka
+
+/// A collection of helper functions wrapping common `rd_kafka_conf_*` functions in Swift.
+struct RDKafkaConfig {
+    /// Wraps a Swift closure inside of a class to be able to pass it to `librdkafka` as an `OpaquePointer`.
+    final class CapturedClosure {
+        typealias Closure = (UnsafePointer<rd_kafka_message_t>?) -> Void
+        let closure: Closure
+
+        init(_ closure: @escaping Closure) {
+            self.closure = closure
+        }
+    }
+
+    /// Create a new `rd_kafka_conf_t` object in memory and initialize it with the given configuration properties.
+    /// - Parameter configDictionary: A dictionary containing the Kafka client configurations.
+    /// - Returns: An `OpaquePointer` pointing to the newly created `rd_kafka_conf_t` object in memory.
+    /// - Throws: A ``KafkaError`` if setting a config value failed.
+    static func createFrom(configDictionary: [String: String]) throws -> OpaquePointer {
+        let configPointer: OpaquePointer = rd_kafka_conf_new()
+        try configDictionary.forEach { key, value in
+            try Self.set(configPointer: configPointer, key: key, value: value)
+        }
+
+        return configPointer
+    }
+
+    /// A Swift wrapper for `rd_kafka_conf_set`.
+    /// - Parameter configPointer: An `OpaquePointer` pointing to the `rd_kafka_conf_t` object in memory.
+    /// - Parameter key: The configuration property to be changed.
+    /// - Parameter value: The new value of the configuration property to be changed.
+    /// - Throws: A ``KafkaError`` if setting the value failed.
+    static func set(configPointer: OpaquePointer, key: String, value: String) throws {
+        let errorChars = UnsafeMutablePointer<CChar>.allocate(capacity: KafkaClient.stringSize)
+        defer { errorChars.deallocate() }
+
+        let configResult = rd_kafka_conf_set(
+            configPointer,
+            key,
+            value,
+            errorChars,
+            KafkaClient.stringSize
+        )
+
+        if configResult != RD_KAFKA_CONF_OK {
+            let errorString = String(cString: errorChars)
+            throw KafkaError.config(reason: errorString)
+        }
+    }
+
+    /// A Swift wrapper for `rd_kafka_conf_set_dr_msg_cb`.
+    /// Defines a function that is called upon every message acknowledgement.
+    /// - Parameter configPointer: An `OpaquePointer` pointing to the `rd_kafka_conf_t` object in memory.
+    /// - Parameter callback: A closure that is invoked upon message acknowledgement.
+    /// - Returns: A ``CapturedClosure`` object that must me retained by the caller as long as acknowledgements are received.
+    static func setDeliveryReportCallback(
+        configPointer: OpaquePointer,
+        _ callback: @escaping ((UnsafePointer<rd_kafka_message_t>?) -> Void)
+    ) -> CapturedClosure {
```

Comment:
Instead of returning a captured closure here we should be able to express this more safe in Swift. Above I already mentioned that I think we should have a method on `RDKafkaConfig` that returns an `rd_kafka`. We should probably have that method return `KafkaClient` instead and add an init to `KafkaClient` that takes both a `rd_kafka` and `CapturedClosure` to make sure this is never dropped.